runtime/src/main/java/io/quarkiverse/messaginghub/pooled/jms/PooledLocalTransactionConnection.java

```diff
@@ -0,0 +1,75 @@
+package io.quarkiverse.messaginghub.pooled.jms;
+
+import javax.jms.Connection;
+import javax.jms.JMSException;
+import javax.jms.Session;
+import javax.transaction.RollbackException;
+import javax.transaction.Status;
+import javax.transaction.Synchronization;
+import javax.transaction.SystemException;
+import javax.transaction.TransactionManager;
+
+import org.messaginghub.pooled.jms.JmsPoolSession;
+import org.messaginghub.pooled.jms.pool.PooledConnection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PooledLocalTransactionConnection extends PooledConnection {
+    private static final Logger LOG = LoggerFactory.getLogger(PooledLocalTransactionConnection.class);
+    private final TransactionManager transactionManager;
+
+    public PooledLocalTransactionConnection(final Connection connection, final TransactionManager transactionManager) {
+        super(connection);
+        this.transactionManager = transactionManager;
+    }
+
+    @Override
+    public Session createSession(boolean transacted, int ackMode) throws JMSException {
+        JmsPoolSession session = (JmsPoolSession) super.createSession(transacted, ackMode);
+
+        try {
+            if (transactionManager != null && transactionManager.getStatus() != Status.STATUS_NO_TRANSACTION
+                    && transacted) {
```

Comment:
OK, I see. It seems that we need to have a config with transaction integration mode.

- **ENABLED**: Integrate JMS Session with Quarkus TransactionManger and force `trasancted=true`
- **XA**: Integrate JMS Session with Quarkus TransactionManager and enable XA support and force`transacted=false`
- **NONE**: Don't integrate with Quarkus TransactionManager and leave `transacted` as-is

Reply:
@mpusg I updated the transaction config. Now you need to use
```
quarkus.pooled-jms.transaction=enabled
```

Reply:
No need properties I guest. 

- XA: Integrate JMS Session with Quarkus TransactionManager and enable XA support and force transacted=false
XA transaction is a distribute transaction system over several microservices. So this is a management of several local transaction several microservices, one local transaction by service. By default, `transacted` session is activated. You can watch it inside the ActiveMQXAConnection class. 
So you can keep your property `pooled-jms.xa.enabled` and authorize a microservice to enter a XA transaction context or not.

- ENABLED: Integrate JMS Session with Quarkus TransactionManger and force transacted=true
Here you haven't needed a property because you know if you're inside a transactional context or not by the `transactionmanager`; ideally during the quarkus build time. 
Basically, if you have a `@Transactional` on a caller, you instanciate `PooledLocalTransactionConnection` and you force the sessions `transacted` to true. Correct.
Be careful of the TxType status, which is an attribute of the @Transactional. If you have `TxType = NOT_SUPPORTED` which mean exclude a treatment from a transaction context, you should instantiate a standard ActiveMQConnexion. 

In summary : 
If you're inside a transactional context (`@Transactional` without `TxType = NOT_SUPPORTED`), 
	use PooledLocalTransactionConnection + transacted=true forced 
else 
	use ActiveMQConnection provided by ActiveMQConnectionFactory. 

- DISABLED: Don't integrate with Quarkus TransactionManager and leave transacted as-is 
`@Transactional (Transactional.TxType.NOT_SUPPORTED)` or no annotation do the job I think.

It will be great if you could analyse the @Transactional during the quarkus build. It could possibly to avoid the following control during the run. 
```
if (transactionManager != null && transactionManager.getStatus() != Status.STATUS_NO_TRANSACTION && transacted)
```

And hard force the transacted like : 
```
JmsPoolSession session = (JmsPoolSession) super.createSession(true, ackMode);
```

Is it clear for you ?

Reply:
Well, `JMS Connection` can be used without `@Transactional` and the users can control it by themselves. e.g.
```
tm.begin();
connectionFactory.getConnection();
...
tm.commit();
```

So I think we have to check the transaction status at runtime. And even if we are running in a transaction, there is still a case that we don't want JMS to involve in it.

- **ENABLED**: we want to involve in a managed transaction but non-XA
- **XA**:  we want to involve in a XA transaction
- **DISABLED**: we don't want to involve in any transaction

Now the default value is **DISABLED** because `quarkus-narayana-jta` is optional.

Reply:
@mpusg can you check it running as expected?

Reply:
Sorry, I didn't understand that you use the same property with three values.

Yes, my code will be like your example. 

But what will be the behavior when you have this following combination ? 
- quarkus.pooled-jms.transaction=enabled and no @Transactional ?
- quarkus.pooled-jms.transaction=disabled and @Transactional on the method ?

For me @Transactional on the method is equivalent of quarkus.pooled-jms.transaction=enabled.

Reply:
> * **ENABLED**: we want to involve in a managed transaction but non-XA

does this mean that in that case you would ignore whatever the value is for param  `transacted` in `Connection.createSession(boolean transacted,
 int acknowledgeMode)`?
cf https://docs.oracle.com/javaee/7/api/javax/jms/Connection.html#createSession-boolean-int-

would I still be possible to manage the transaction at the jms level if I wanted for one particular use case?

Reply:
> quarkus.pooled-jms.transaction=enabled and no `@Transactional`?

The message will send as-is

> quarkus.pooled-jms.transaction=disabled and `@Transactional` on the method ?

The JMS connection does not involve in a transaction even if it is running. That means if there is any Runtime Exception during the method invoking and TransactionManger decides to **rollback** it, the message is still sending and never be rollbacked.

`@Transactional` is an annotation which defines the border of the transaction. By using a `Interceptor`, it starts a transaction at the beginning of the method invoking and decides to commit or rollback it in the end. But it does not guarantee all of the operations (JMS, JDBC, ...) must be involved in.

Reply:
@vsevel 
> does this mean that in that case you would ignore whatever the value is for param transacted?

Yeah, we force to set `transacted` to `true`  because we need to call `session.commit()` or `session,rollback()` at the enf of transction.

> would I still be possible to manage the transaction at the jms level if I wanted for one particular use case?

I think so but it could cause the transaction failure. e.g. you call `session.rollback()` during the bussiness method, but in the end of transaction, TransactionManager decides to commit and it will fail.

Reply:
if we have:
```
@Transactional(NOT_SUPPORTED)
void doJMS() {
  Session session = connection.createSession(true, ...)
  ...
  Message msg = consumer.receive()
  doBusiness(msg)
  session.commit()
}

@Transactional(REQUIRED)
void doBusiness(Message msg) {
  // ...
}
```

that should work, right?
you are not going to register the session with a non existing transaction?

Reply:
Yeah, exactly. It should work.

Reply:
ok we are going to run some more tests. thanks for your answers